src/SurveyService.Data/QuestionRepository.cs

```diff
@@ -11,6 +13,25 @@ public class QuestionRepository : IQuestionRepository
 {
   private readonly IDataProvider _provider;
 
+  private IQueryable<DbQuestion> CreateGetPredicates(
+    GetQuestionFilter filter,
+    IQueryable<DbQuestion> dbQuestions)
+  { 
+    if (filter.IncludeAnswers)
+    {
+      dbQuestions = dbQuestions
+        .Include(question => question.Options.Where(option => option.IsActive))
+          .ThenInclude(option => option.UsersAnswers);
+    }
+    else
+    {
+      dbQuestions = dbQuestions
+        .Include(question => question.Options.Where(option => option.IsActive));
```

Comment:
if includeOptions=true and includeAnswers=false include only active and **not-custom** options